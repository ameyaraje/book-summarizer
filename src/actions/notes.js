import { v4 as uuidv4 } from 'uuid';

export const addNote = ({
    title = '',
    author = '',
    isbn = '',
    text = ''
} = {}) => {
    return {
        type: 'ADD_NOTE',
        note: {
            id: uuidv4(),
            title,
            author,
            isbn,
            text
        }
    };
};

export const removeNote = ({ id } = {}) => {
    return {
        type: 'REMOVE_NOTE',
        id: id
    };
};

export const editNote = (id, updates) => {
    return {
        type: 'EDIT_NOTE',
        id: id,
        updates: updates
    };
};

export const setNotes = (notes) => {
    return {
        type: 'SET_NOTES',
        notes: notes
    };
};