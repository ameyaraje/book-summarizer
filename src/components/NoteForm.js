import React, { useEffect, useState } from 'react';

const NoteForm = (props) => {
    // console.log(props);

    const [title, setTitle] = useState('');
    const [author, setAuthor] = useState('');
    const [isbn, setIsbn] = useState('');
    const [bookText, setBookText] = useState('');
    const [error, setError] = useState({
        state: false,
        message: ''
    });

    useEffect(() => {
        if (props.book) {
            setTitle(props.book.title);
            setAuthor(props.book.author);
            setBookText(props.book.text);
            setIsbn(props.book.isbn);
        }
    }, [props.book]);

    const onAuthorChanged = (e) => {
        setAuthor(e.target.value);
    };

    const onTitleChanged = (e) => {
        setTitle(e.target.value);
    };

    const onIsbnChanged = (e) => {
        setIsbn(e.target.value);
    };

    const onBookTextChanged = (e) => {
        setBookText(e.target.value);
    };

    const onFormSubmit = (e) => {
        e.preventDefault();

        if (title === '' || author === '' || isbn === '') {
            setError({
                state: true,
                message: 'Please fill out the missing fields'
            });
        // } else if (!isbn.match(`^(?:ISBN(?:-13)?:?\ )?(?=[0-9]{13}$|(?=(?:[0-9]+[-\ ]){4})[-\ 0-9]{17}$)97[89][-\ ]?[0-9]{1,5}[-\ ]?[0-9]+[-\ ]?[0-9]+[-\ ]?[0-9]$`)) {
        //     setError({
        //         state: true,
        //         message: 'Please verify the ISBN entered'
        //     });
        } else {
            props.onSubmit({
                author: author,
                title: title,
                isbn: isbn,
                text: bookText,
            });
        }
    };

    return (
        <div>
            <h3>
                Note Form
            </h3>
            {
                error.state && <h4>{error.message}</h4>
            }
            <form>
                <input type="text" placeholder="Author" autoFocus value={author} onChange={onAuthorChanged} />
                <input type="text" placeholder="Title" value={title} onChange={onTitleChanged} />
                <input type="text" placeholder="ISBN" value={isbn} onChange={onIsbnChanged} />
                <textarea cols="30" rows="10" placeholder="Enter Book Notes" value={bookText} onChange={onBookTextChanged}></textarea>
                <button onClick={onFormSubmit}>Submit</button>
            </form>
        </div>
    );
};

export default NoteForm;