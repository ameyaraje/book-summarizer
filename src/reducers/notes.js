// const demoState = {
//     notes: [{
//         id: 'dsfdsf',
//         title: 'abcd',
//         author: 'asdasd',
//         isbn: 'sdfdsfdf',
//         text: 'esdfdf'
//     }],
//     filters: {
//         author: 'dsfsdf',
//         title: 'asdas'
//     }
// };

const notesReducerDefaultState = [];

const notesReducer = (state = notesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_NOTE':
            return [
                ...state,
                action.note
            ];

        case 'REMOVE_NOTE':
            return state.filter(({ id }) => {
                return id !== action.id
            });

        case 'EDIT_NOTE':
            return state.map((note) => {
                if (note.id === action.id)
                    return {
                        ...note,
                        ...action.updates
                    };
                else
                    return note;
            });

        case 'SET_NOTES':
            return action.notes;

        default:
            return state;
    }
};

export default notesReducer;
