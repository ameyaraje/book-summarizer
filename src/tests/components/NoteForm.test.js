import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';

import NoteForm from '../../components/NoteForm';
import notes from '../fixtures/notes';

test('should render NoteForm correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<NoteForm />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render NoteForm with some data', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<NoteForm book={notes[1]} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

// test('should render error for invalid form submission', () => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<NoteForm />);

// });