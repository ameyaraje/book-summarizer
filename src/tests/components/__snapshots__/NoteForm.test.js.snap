// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NoteForm correctly 1`] = `
<div>
  <h3>
    Note Form
  </h3>
  <form>
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Author"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="ISBN"
      type="text"
      value=""
    />
    <textarea
      cols="30"
      onChange={[Function]}
      placeholder="Enter Book Notes"
      rows="10"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should render NoteForm with some data 1`] = `
<div>
  <h3>
    Note Form
  </h3>
  <form>
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Author"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="ISBN"
      type="text"
      value=""
    />
    <textarea
      cols="30"
      onChange={[Function]}
      placeholder="Enter Book Notes"
      rows="10"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
